sql: remove the list of descriptor IDs from the output of SHOW JOBS.

knz opened the pull request:
Since descriptor IDs are not (yet) part of the public UI of
CockroachDB, they should not be printed out by documented statements
like SHOW.

Suggested in https://github.com/cockroachdb/docs/pull/1872#issuecomment-332006659.
Resolves the discussion on that issue.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18771)
<!-- Reviewable:end -->

benesch commented on pkg/server/admin.go:
```diff
@@ -1032,9 +1032,9 @@ func (s *adminServer) Jobs(
 	q := makeSQLQuery()
 	q.Append(`
-			SELECT id, type, description, username, descriptor_ids, status,
+			SELECT id, type, description, username, status, descriptor_ids,
```
Why the reordering?

benesch commented on pkg/sql/jobs/jobs.go:
```diff
@@ -726,7 +726,10 @@ func RunAndWaitForTerminalState(
+		err := sqlDB.QueryRow(
+			`SELECT status, error, fraction_completed
+         FROM crdb_internal.jobs
+        WHERE id = $1`, jobID).Scan(
```
Since this doesn't use any internal columns, I'd prefer to leave it as `SHOW JOBS` so we actually exercise the `SHOW JOBS` code path somewhere.

benesch replied:
Also, I'm not sure I understand your indentation scheme.

benesch commented on pkg/sql/show.go:
```diff
@@ -601,7 +601,11 @@ func (p *planner) ShowQueries(ctx context.Context, n *parser.ShowQueries) (planN
+	return p.delegateQuery(ctx, "SHOW JOBS",
+		`SELECT id, type, description, username, status, created, started, finished, modified,
+            fraction_completed, error, coordinator_id
+       FROM crdb_internal.jobs`,
```
Ditto here—the spacing seems off.

benesch commented on pkg/sql/show_test.go:
```diff
@@ -542,21 +542,19 @@ func TestShowJobs(t *testing.T) {
+	)
+	if !reflect.DeepEqual(in, out) {
+		diff := strings.Join(pretty.Diff(in, out), "\n")
+		t.Fatalf("in job did not match out job:\n%s", diff)
```
It makes me somewhat nervous to stop testing `SHOW JOBS`, but I don't have a great alternative.

benesch reviewed (commented):
Thanks for doing this, @knz!

knz commented:
Review status: 0 of 6 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[pkg/server/admin.go, line 1035 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-KuwHnaIfwsQsCSCBgb5-r1-1035:-Kuz66751KPitjJBa6CE:b-v9dvhy) ([raw file](https://github.com/cockroachdb/cockroach/blob/418be1f579876cf9163199ea44d69a6d212bddf0/pkg/server/admin.go#L1035)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Why the reordering?
</blockquote></details>

My mistake. Reverted.

---

*[pkg/sql/show.go, line 607 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-KuwHnaIfwsQsCSCBgb8-r1-607:-Kuz67VEwnq2160purEf:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/418be1f579876cf9163199ea44d69a6d212bddf0/pkg/sql/show.go#L607)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Ditto here—the spacing seems off.
</blockquote></details>

Fixed.

---

*[pkg/sql/show_test.go, line 558 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-KuwHnaIfwsQsCSCBgb9-r1-558:-Kuz69qAsOq1halEaRuF:bweqg6l) ([raw file](https://github.com/cockroachdb/cockroach/blob/418be1f579876cf9163199ea44d69a6d212bddf0/pkg/sql/show_test.go#L558)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

It makes me somewhat nervous to stop testing `SHOW JOBS`, but I don't have a great alternative.
</blockquote></details>

Alas, me neither.

---

*[pkg/sql/jobs/jobs.go, line 732 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-KuwHnaIfwsQsCSCBgb6-r1-732:-Kuz6n6cRAV4zVvPS1uq:bbglb15) ([raw file](https://github.com/cockroachdb/cockroach/blob/418be1f579876cf9163199ea44d69a6d212bddf0/pkg/sql/jobs/jobs.go#L732)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Also, I'm not sure I understand your indentation scheme.
</blockquote></details>

Standard SQL intendation scheme: the keywords of a SELECT are right-aligned:


```
   SELECT foo, bar, baz
     FROM blip, blip
    WHERE foo > bar
      AND bar > baz
 ORDER BY baz
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771)*
<!-- Sent from Reviewable.io -->

benesch commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

Thanks, @knz!

---

Reviewed 2 of 5 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/jobs/jobs.go, line 732 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-KuwHnaIfwsQsCSCBgb6-r1-732:-Kuz7hgjmFRMC-Vq6CSf:b7joi34) ([raw file](https://github.com/cockroachdb/cockroach/blob/418be1f579876cf9163199ea44d69a6d212bddf0/pkg/sql/jobs/jobs.go#L732)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Standard SQL intendation scheme: the keywords of a SELECT are right-aligned:


```
   SELECT foo, bar, baz
     FROM blip, blip
    WHERE foo > bar
      AND bar > baz
 ORDER BY baz
```
</blockquote></details>

Aha! TIL. Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18771#-:-Kuz7o3HE0QdjjcZwrim:batqzwa)*
<!-- Sent from Reviewable.io -->

knz committed:
sql: remove the list of descriptor IDs from the output of SHOW JOBS.

Since descriptor IDs are not (yet) part of the public UI of
CockroachDB, they should not be printed out by documented statements
like SHOW.

knz commented:
TFYR :purple_heart:
